src/services/open-api-client.js

```diff
@@ -34,28 +35,22 @@ class OpenApiClient {
     const params = this.getParams(opts, operation);
 
     if (!opts.uri) {
-      opts.uri = this.getUri(opts);
+      opts.uri = this.getUri(opts.uri, opts);
     }
 
     // If the URI is relative, determine the host and prepend it.
     if (opts.uri.startsWith('/')) {
       if (!opts.host) {
         opts.host = path.server;
       }
-
-      if (opts.region) {
-        const parts = opts.host.split('.');
-
-        // From 'https://api.twilio.com/' to 'https://api.{region}.twilio.com/'
-        if (parts.length > 1 && parts[1] !== opts.region) {
-          parts.splice(1, 0, opts.region);
-          opts.host = parts.join('.');
-        }
-      }
-
       opts.uri = opts.host + opts.uri;
     }
 
+    const uri = new url.URL(opts.uri);
+    uri.hostname = this.getHost(uri.hostname, opts);
+    uri.pathname = this.getUri(uri.pathname, opts);
```

Comment:
I don't think this line is needed (along with the changes to `getUri`). The path should already be fully-formed at this point.

Reply:
Hmm.. when I take it out it breaks the tests..

Reply:
Is there a test passing a parameterized `opts.uri`?

Reply:
Not a parameterized `opts.uri`, but we have a parameterized `opts.path`

Reply:
Line 38 should eval the `opts.path` there.